### PR TITLE
feat: measure cdn stats (DTCRCGEMI-194)

### PR DESCRIPTION
**Pull request by jadutter**
Measure the time it takes for an iframe to load its first script,

### CODE REVIEW

File: src/utils/performance.js [hunk 1]
```diff
@@ -2,7 +2,48 @@ import { getPerformance, isIE } from 'belter/src';
 
 const namespaced = name => `__paypal_messaging_performance__${name}`;
 
-const performance = getPerformance();
+export const performance = getPerformance();
+
+export const PERFORMANCE_MEASURE_KEYS = {
```
Comment (jadutter) [inline src/utils/performance.js:7]: I wanted to make it more clear on what we are tracking by ensuring the items are enumerated here. 

While checking what we are tracking, I noticed some things were listed as deprecated and made a note in the comments. 

This map does not include the renderStart metric we are collecting. I would have liked to have refactored it to be included in this map, but I needed to begin work on the PR for cpnw.

File: src/utils/performance.js [hunk 2]
```diff
@@ -2,7 +2,50 @@ import { getPerformance, isIE } from 'belter/src';
...
+            ({ name, entryType }) =>
+                (entryType === 'navigation' && `${name}`.indexOf('/credit-presentment/smart/message') > -1) ||
+                (entryType === 'resource' && `${name}`.indexOf('/credit-presentment/renderMessage') > -1)
+        );
+
+    const [{ connectStart, responseStart }] = requests.slice(-1);
+
+    if (typeof connectStart !== 'undefined') {
+        // This measures the "Waiting (Time To First Byte)" for the request;
+        // how long we've spent waiting for a response after sending the request
```
Comment (jadutter) [inline src/utils/performance.js]: The value recorded here will match the value in
> Dev Tools > Network > Select the request > Timing > Waiting (TTFB)

File: src/utils/performance.js [hunk 3]
```diff
@@ -2,7 +2,50 @@ import { getPerformance, isIE } from 'belter/src';
...
+    // eslint-disable-next-line compat/compat
+    const requests = window.performance
+        .getEntries()
+        .filter(
+            ({ name, entryType }) =>
+                (entryType === 'navigation' && `${name}`.indexOf('/credit-presentment/smart/message') > -1) ||
+                (entryType === 'resource' && `${name}`.indexOf('/credit-presentment/renderMessage') > -1)
+        );
+
+    const [{ connectStart, responseStart }] = requests.slice(-1);
```
Comment (gillycheesesteak) [inline src/utils/performance.js]: This could cause a destructuring error if `requests` is somehow empty

File: src/utils/performance.js [hunk 4]
```diff
@@ -11,12 +54,12 @@ export function getPerformanceMeasure(name) {
 export function getNavigationTiming(name) {
     const entry = performance?.getEntriesByType('navigation')[0];
 
-    return entry?.[name] ?? -1;
+    return Math.round(entry?.[name] ?? -1).toString();
```
Comment (gillycheesesteak) [inline src/utils/performance.js]: should this use the new `formatStat` function?

File: src/utils/performance.js (hunk 2 above)
Comment (gillycheesesteak) [reply src/utils/performance.js]: Is `responseEnd` available here? That would encompass the time to download the content as well

**Review by gillycheesesteak — APPROVED** [review]

**Pull request merged**